backend/controller/ingress/handler.go

```diff
@@ -35,13 +37,16 @@ func Handle(
 			http.NotFound(w, r)
 			return
 		}
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		logger.Errorf(err, "failed to resolve route for %s %s", r.Method, r.URL.Path)
+		http.Error(w, internalServerError, http.StatusInternalServerError)
 		return
 	}
 
 	body, err := BuildRequestBody(route, r, sch)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
+		// Only log at debug, as this is a client side error
+		logger.Debugf("bad request: %s", err.Error())
+		http.Error(w, internalServerError, http.StatusBadRequest)
```

Comment:
Actually, this is the error case where we will want to return some subset of the underlying errors. The validation errors returned are intended for the caller.

Reply:
good point